Split/FetcherEngine/Cache/SplitCache.swift

```diff
@@ -83,6 +83,14 @@ class SplitCache: SplitCacheProtocol {
     func setTimestamp(timestamp: Int) {
         inMemoryCache.setTimestamp(timestamp: timestamp)
     }
+
+    func kill(splitName: String, defaultTreatment: String, changeNumber: Int64) {
+        if let split = inMemoryCache.getSplit(splitName: splitName) {
+            split.killed = true
```

Comment:
Some considerations:
- shouldn't this logic be on `InMemorySplitCache`?
- shouldn't we check that `split.changeNumber < changeNumber` before updating the split?
- besides, to "persist" the change in the storage, I am not sure in iOS client, but don't we need to call `self.addSplit(splitName, split)` after updating the split?
- finally, consider returning a boolean value wether the split was actually killed or not (for example, it would return false if the splitName doesn't exist or the changeNumber was older). But I understand that it is not used yet in iOS client (in JS the boolean result is used to notify an SDK_UPDATE event to the user).

Reply:
Good catch! Fixed!